pandas/io/pytables.py

```diff
@@ -3148,9 +3148,13 @@ def write_multi_index(self, key: str, index: MultiIndex) -> None:
         ):
             # write the level
             if isinstance(lev.dtype, ExtensionDtype):
-                raise NotImplementedError(
-                    "Saving a MultiIndex with an extension dtype is not supported."
-                )
+                # GH 63412
+                if isinstance(lev.dtype, StringDtype):
+                    lev = lev.astype(object)
```

Comment:
If you replace this with `pass`, does `_convert_index` below already convert this data correctly without the `astype`?